attestation-service/attestation-service/src/lib.rs

```diff
@@ -109,29 +71,55 @@ impl AttestationService {
             .map_err(|e| anyhow!("Cannot Set Policy: {:?}", e))
     }
 
+    fn accumulate_hash(materials: &[Vec<u8>]) -> Option<Vec<u8>> {
+        if materials.is_empty() {
+            return None;
+        }
+        let mut hasher = Sha384::new();
+        materials.iter().for_each(|m| hasher.update(m));
+        Some(hasher.finalize().to_vec())
+    }
+
     /// Evaluate Attestation Evidence.
     /// Issue an attestation results token which contain TCB status and TEE public key.
     pub async fn evaluate(&self, tee: Tee, nonce: &str, attestation: &str) -> Result<String> {
         let attestation = serde_json::from_str::<Attestation>(attestation)
             .context("Failed to deserialize Attestation")?;
-        let verifier = crate::verifier::to_verifier(&tee)?;
+        let verifier = verifier::to_verifier(&tee)?;
+
+        let report_data = Self::accumulate_hash(&[
+            nonce.as_bytes().to_vec(),
+            attestation.tee_pubkey.k_mod.as_bytes().to_vec(),
+            attestation.tee_pubkey.k_exp.as_bytes().to_vec(),
+        ]);
 
         let claims_from_tee_evidence = verifier
-            .evaluate(nonce.to_string(), &attestation)
+            .evaluate(
+                attestation.tee_evidence.as_bytes(),
+                report_data.as_deref(),
+                // We currently do not need to check the initdata hash.
+                // Because in KBS protocol, the hash(nonce, tee_key) is binded
```

Comment:
```suggestion
                // Because hash(nonce, tee_key) is bound to the `report_data` in the KBS protocol.
```

Reply:
btw, I cannot follow this comments, how the `report_data` which is hash(nonce, tee_key) impacts initdata hash, shouldn't the initdata something like a `HOSTDATA` for snp or `cca-realm-personalization-value` in CCA?

Reply:
Ah. Yes. I just want to say that we do not use `initdata hash` field of `verifier` crate. We will use this later.

Reply:
Let make the notes readable.

Reply:
okay, git it, so you might just say this field it not available until the refactor is done?

Reply:
Yes. It cannot be exposed by CoCo-AS now, **but can** be directly use by import `verifier` crate.